docs/usage/configuration-options.md

```diff
@@ -1809,6 +1809,27 @@ It uses `QuickLRU` with a `maxSize` of `1000`.
 
 Enable got [http2](https://github.com/sindresorhus/got/blob/v11.5.2/readme.md#http2) support.
 
+### header
+
+You can provide `header` object that includes fields to be forwarded to the HTTP request header. By default, all header starting with "X-" are allowed, a bot administrator may configure an override for [allowedHeader](./self-hosted-configuration.md#allowedHeader) to configure more permitted header.
+
+Any `header` value configured in bot admin `hostRules` (e.g. `config.js`) won't be validated so can contain any desired header regardless of `allowedHeader`.
```

Comment:
```suggestion
You can provide a `header` object that includes fields to be forwarded to the HTTP request header.
By default, all headers starting with "X-" are allowed.

A bot administrator may configure an override for [`allowedHeader`](./self-hosted-configuration.md#allowedHeader) to configure more permitted headers.

`header` value(s) configured in the bot admin `hostRules` (for example in a `config.js` file) are _not_ validated, so it may contain any header regardless of `allowedHeader`.
```

Some small improvements to start with.

The docs for the `header` config option should explain _why_ somebody would want to override the default headers. It's not clear after reading this text what I as end-user gain from this option, when I should use it, when I should _not_ use it, if there's anything else I should know about.